test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -603,4 +610,146 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
+		})
+
+		ginkgo.AfterEach(func() {
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, onDemandFlavor, true)
+		})
+
+		ginkgo.It("Should update of the pending workloads when a new workload is scheduled", func() {
+			workloads := []*kueue.Workload{
+				testing.MakeWorkload("one", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "2").Request(resourceGPU, "2").Obj(),
+				testing.MakeWorkload("two", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "3").Request(resourceGPU, "3").Obj(),
+			}
+
+			ginkgo.By("Creating workloads")
+			for _, w := range workloads {
+				gomega.Expect(k8sClient.Create(ctx, w)).To(gomega.Succeed())
+			}
+			gomega.Eventually(func() *kueue.ClusterQueuePendingWorkloadsStatus {
+				var updatedCq kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCq)).To(gomega.Succeed())
+				return updatedCq.Status.PendingWorkloadsStatus
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&kueue.ClusterQueuePendingWorkloadsStatus{
+				Head: []kueue.ClusterQueuePendingWorkload{
+					{
+						Name:      "one",
+						Namespace: ns.Name,
+					},
+					{
+						Name:      "two",
+						Namespace: ns.Name,
+					},
+				},
+			},
+				ignoreLastChangeTime,
+				cmpopts.SortSlices(func(i, j kueue.ClusterQueuePendingWorkload) bool {
+					return i.Name < j.Name
+				})))
+
+			ginkgo.By("Creating a new workload")
+			newWorkloads := []*kueue.Workload{
+				testing.MakeWorkload("three", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "2").Request(resourceGPU, "2").Obj(),
+				testing.MakeWorkload("four", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "3").Request(resourceGPU, "3").Obj(),
+			}
+			for _, w := range newWorkloads {
+				gomega.Expect(k8sClient.Create(ctx, w)).To(gomega.Succeed())
+				time.Sleep(time.Second)
+			}
+
+			ginkgo.By("Cut off the head of pending workloads when the number of pending workloads exceeds MaxCount")
+			gomega.Eventually(func() *kueue.ClusterQueuePendingWorkloadsStatus {
+				var updatedCq kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCq)).To(gomega.Succeed())
+				return updatedCq.Status.PendingWorkloadsStatus
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&kueue.ClusterQueuePendingWorkloadsStatus{
+				Head: []kueue.ClusterQueuePendingWorkload{
+					{
+						Name:      "one",
+						Namespace: ns.Name,
+					},
+					{
+						Name:      "two",
+						Namespace: ns.Name,
+					},
+					{
+						Name:      "three",
+						Namespace: ns.Name,
+					},
+				},
+			}, ignoreLastChangeTime, cmpopts.SortSlices(func(i, j kueue.ClusterQueuePendingWorkload) bool {
+				return i.Name < j.Name
+			})))
+
+			ginkgo.By("Admitting workloads")
+			admissions := []*kueue.Admission{
```

Comment:
I think it will be more readable to admit manually workloads, rather than by the loop. I think it is enough to just admit workload 1 and 2 for the purpose of the testing.

Reply:
loop was removed https://github.com/kubernetes-sigs/kueue/commit/3f01af197b7d150455e4421a70ae1c12cf59c0e6